tools/Datadog.Core.Tools/ConcurrencyHelper.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Datadog.Core.Tools
+{
+    public class ConcurrencyHelper
+    {
+        private readonly ManualResetEventSlim _initiateLevelsEvent = new ManualResetEventSlim(initialState: false); // Start blocked
+        private readonly ManualResetEventSlim _testFinishedEvent = new ManualResetEventSlim(initialState: false); // Start blocked
+        private int _remainingLevels;
+
+        public enum HelperStatus
+        {
+            AwaitingConfig,
+            Ready,
+            Running,
+            Errored,
+            Finished
+        }
+
+        public List<Level> Levels { get; set; } = new List<Level>();
+
+        public HelperStatus Status { get; private set; } = HelperStatus.AwaitingConfig;
+
+        public DateTime Started { get; private set; }
+
+        public DateTime Finished { get; private set; }
+
+        public void RegisterLevel(Action action, int iterations, string friendlyName = null, int numberOfRegisters = 1)
+        {
+            while (numberOfRegisters-- > 0)
```

Comment:
This syntax seems a bit unusual, especially with the postfix operator. Wouldn't a simple `for` loop be more readable?

```suggestion
            for (int x = 0; x < numberOfRegisters; x++)
```

Reply:
Sure, I opted for not having the extra variable, but it's not important, this is more readable for sure.